### DISCUSSION TITLE
Rec.Next() = 0 vs Rec.Next() < 0

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#399] [New Rule]
Is there a benefit on using `Next() < 1` versus `Next() = 0`?

```AL
procedure MyProcedure()
var
    Customer: Record Customer;
begin
    Customer.FindSet();
    repeat
        ...
    until Customer.Next() < 1;
end;
```

```AL
  procedure MyProcedure()
  var
      Customer: Record Customer;
  begin
      Customer.FindSet();
      repeat
          ...
      until Customer.Next() = 0;
  end;
```

For consistency sake, would it make sense to make a rule for this?

**Comment by StefanMaron** [comment]
If you call Next() without parameters, there are exactly two possible return values `0` - No record left/end of table and `1` - moved to the next record

If you do specify a parameter for Next() the turn value will be different. In that case you should also know what return value to expect and what to test for.

I know where the `<1` approach comes from. But in my opinion its total bullsh*** :)

**Reply by Arthurvdv** [reply]
> If you do specify a parameter for Next() the turn value will be different.

Ah, now I remember that was the case to somehow trigger an outcome other then `0` or `1`.
So the `Next() = 0` is the preferred choice?

**Reply by StefanMaron** [reply]
you can use next(-1)  to go backwards, and then the return value would also be -1

**Reply by Arthurvdv** [reply]
🤯

I'm getting the idea that creating a rule here is not a good idea. What do you think?
